app/frontend/src/api/api.ts

```diff
@@ -32,23 +37,58 @@ export async function askApi(request: ChatAppRequest, idToken: string | undefine
     return parsedResponse as ChatAppResponse;
 }
 
-export async function configApi(idToken: string | undefined): Promise<Config> {
-    const response = await fetch(`${BACKEND_URI}/config`, {
-        method: "GET",
-        headers: getHeaders(idToken)
-    });
-
-    return (await response.json()) as Config;
-}
-
 export async function chatApi(request: ChatAppRequest, idToken: string | undefined): Promise<Response> {
     return await fetch(`${BACKEND_URI}/chat`, {
         method: "POST",
-        headers: getHeaders(idToken),
+        headers: { ...getHeaders(idToken), "Content-Type": "application/json" },
         body: JSON.stringify(request)
     });
 }
 
 export function getCitationFilePath(citation: string): string {
     return `${BACKEND_URI}/content/${citation}`;
 }
+
+export async function uploadFileApi(request: FormData, idToken: string): Promise<SimpleAPIResponse> {
+    const response = await fetch("/upload", {
+        method: "POST",
+        headers: getHeaders(idToken),
+        body: request
+    });
+
+    if (!response.ok) {
+        throw new Error(`Uploading files failed: ${response.statusText}`);
+    }
+
+    const dataResponse: SimpleAPIResponse = await response.json();
+    return dataResponse;
+}
+
+export async function deleteUploadedFileApi(filename: string, idToken: string): Promise<SimpleAPIResponse> {
+    const response = await fetch("/delete_uploaded", {
+        method: "POST",
+        headers: { ...getHeaders(idToken), "Content-Type": "application/json" },
+        body: JSON.stringify({ filename })
```

Comment:
I'm actually not familiar with this syntax. does it expand to a object like this?
{ "filename": filename }

Reply:
Yep, see here:
https://leanpub.com/understandinges6/read/#leanpub-auto-object-literal-syntax-extensions